Update Predator Bot

Sparks1992 opened the pull request:
Added Stone Hide skill usage after using Venomous Leap skill
Added Stone Hide skill usage when retreating

Anakonda commented on bots/heroes/predator_main.lua:
```diff
@@ -276,6 +277,8 @@ local function HarassHeroExecuteOverride(botBrain)
 				if bDebugEchos then BotEcho('LEAPIN!') end
 				bActionTaken = core.OrderAbilityEntity(botBrain, leap, unitTarget)
+					if stonehide:CanActivate() then
+						bActionTaken = core.OrderAbility(botBrain, stonehide) 
```
This if is missing end. That is syntax error - do you even test these?

Anakonda commented on bots/heroes/predator_main.lua:
```diff
@@ -288,7 +291,14 @@ end
+--this is a great function to override with using retreating skills, such as blinks, travels, stuns or slows.
+function behaviorLib.CustomReturnToWellExecute(botBrain)
+	local stonehide = skills.abilStoneHide
+	return  core.OrderAbility(botBrain, stonehide)
```
Check if it can be activated, also only cast it if there are enemies

Sparks1992 commented:
Changed both issues:
- added the end (sorry, can't test them really well since there is no test+++ button in practice with bots)
- added bCanSee to retreat use of StoneHide

Anakonda commented:
open console with ctrl+f8 run command ´showwidget testppp_mainButton´ and close console with ctrl+f8. This should give you the test+++ button.
unitTarget is the hero you last tried to kill it can be anywhere in the map by the time you are retreating - like visible on another lane. You need to use the localunits table, see Moon Queens ult if you need exampl.e

Sparks1992 commented:
I updated it, i tried to add Predator in bots folder but i think it's overwritten by the one from the game, because in the bots pick it appears only one Predator, and that's the one from the game.

Sparks1992 commented:
done, it works...but it activates too late...first he retreats without using the skill and he starts using it only below 15-20% after i chase him for a couple of seconds. How could i make him use it as soon as he is starting to retreat?

Anakonda commented on bots/heroes/predator_main.lua:
```diff
@@ -288,7 +292,47 @@ end
+	if unitSelf:GetHealthPercent() < .50 then
+		local stonehide = skills.abilStoneHide
+		-- BACK OFF!
+		if bCanSeeUnit and stonehide:CanActivate() then
```
instead of bCanSeeunit check that nCount > 0
Then remove all unitTarget stuff

Sparks1992 commented:
Updated - removed stuff that wasn't used and added nCount > 0 instead of bCanSee
